### DISCUSSION TITLE
Port after launch still 10000

### DISCUSSION THREAD

**Discussion by xKwqRtL** [Budibase/budibase#4338] [Help]
Hi. I used for my budibase project the digitalocean server. I connected my subdomain and everything works fine.

- But the problem is, that the project is only available in the sub.domain.com:10000 port.
- I want to bring it to the sub.domain.com (or sub.domain.com:443).
- I've tried to edit the .env file in the root sftp -> MAIN_PORT=443 (also tried 80)
- I've also tried to open the ports in digitalocean firewall.

How can I change the port to the normal subdomain?

Thanks!
Alex

PS: I really love budibase <3

**Comment by mjashanks** [comment]
Hey Alex, Budibase uses Envoy as a proxy. It's actually envoy that runs on port 10000, and directs requests to the various parts of the stack.

Have a look in `envoy.yaml` - port 10000 should be configured in there.

**Comment by xKwqRtL** [comment]
Thank you! I got it already. But another question regarding https.

In my Caddyfile is my domain on line 1 (like in the doc)
And if I run the command "caddy reverse-proxy --to 46.101.152.181:10000" or "caddy reverse-proxy —from 46.101.152.181:10000 —to localhost:10000"

I get this error (image attached)

What did I do wrong? I did everything like in the docs https://docs.budibase.com/self-hosting/hosting-settings/advanced-options/reverse-proxy#simple-https-setup-with-caddy

Thanks!
Alex
Am 7. Feb. 2022, 11:20 +0100 schrieb Michael Shanks ***@***.***>:

**Reply by shogunpurple** [reply]
Hi @pichlalex2001 - your caddy `reverse-proxy` command is not correct.

As in the docs, just replace the `from` flag with the domain you added to your `Caddyfile`:

`caddy reverse-proxy --from yourdomain.com --to localhost:10000`

**Comment by xKwqRtL** [comment]
Hi

Thanks for your message. I added my domain „my.emplolio.com“ into my Caddyfile in line 1. (Attachment)
Thats what I typed into my command line: caddy reverse-proxy --from my.emplolio.com --to localhost:10000

And that’s still the same error… (Attachment)
Am 7. Feb. 2022, 16:38 +0100 schrieb Martin McKeaveney ***@***.***>:

**Reply by shogunpurple** [reply]
Hey @pichlalex2001 - I can't see any attachment, in either of your comments. Could you attach it here?

**Reply by xKwqRtL** [reply]
<img width="386" alt="Caddyfile" src="https://user-images.githubusercontent.com/20849382/152821800-f4799227-337d-41fc-a305-06b0f36f3fa7.png">
<img width="1327" alt="error2" src="https://user-images.githubusercontent.com/20849382/152821871-cb600a23-f3ae-4d7b-867f-ea6bca30a09a.png">

**Reply by hendrikjap** [reply]
https://github.com/Budibase/budibase/discussions/3853#discussioncomment-1873524

**Reply by hendrikjap** [reply]
Depending on how you install Budibase (1 click or through linux deployment), you must allow certain firewall for Caddy to works properly. Having custom domain DNS managed through digital ocean also may cause problem. You can manage your DNS outside digital ocean.

**Comment by xKwqRtL** [comment]
Yes of course - added to your comment.
Am 7. Feb. 2022, 16:44 +0100 schrieb Martin McKeaveney ***@***.***>:

**Comment by xKwqRtL** [comment]
I have installed the Budibase via One Click Deployment to Digital Ocean. Which Firewall Ports do I have to create for Caddy? I did not find ones.
Without the custom DNS caused the same problem.
Am 8. Feb. 2022, 13:30 +0100 schrieb ely2727 ***@***.***>:

**Reply by shogunpurple** [reply]
You will need to enable port 80 and port 443.

you can do this in ufw by running:
```
sudo ufw allow http
sudo ufw allow https

**Comment by xKwqRtL** [comment]
Okay, done.

I then run the "caddy reverse-proxy —from domain --to localhost:10000“ again.

Still the same issue.
Am 8. Feb. 2022, 14:36 +0100 schrieb Martin McKeaveney ***@***.***>:

**Reply by shogunpurple** [reply]
From your error above, you already have something running on port 443. You need to kill whatever is running on that port.
